### ISSUE TITLE
issue: if 'isValid' value is used, mode and revalidationMode behaves like 'onChange'

### ISSUE THREAD

**Issue by 0x2509** [react-hook-form/react-hook-form#10285]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-basic-forked-5gi2ng

### Steps to reproduce

Based on the closed issue here https://github.com/react-hook-form/react-hook-form/issues/10215 i try to "reopen" this bug.  
It makes no sense, that this behaviour of 'isValid' has changed and that the issue was closed with the answer (https://github.com/react-hook-form/react-hook-form/issues/10215#issuecomment-1492946013)

### Expected behaviour

isValid should not change useForm modes to onChange

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Expected behavior, what's your use case? Why is this an issue?

**Comment by 0x2509** [comment]
> Expected behavior, what's your use case? Why is this an issue?

For example we have 2 async validation calls. So on every onChange they are fired.  
I know i could use debouncing here but is this the best practice in my case?  
So if we use the 'isValid' prop, we are stuck with validation onChange atm and as the tomchen said in the linked issue above, this wasnt always the case and changed with v7.39.  
Such a change should really be documentend.

**Comment by bluebill1049** [comment]
Please refer to the release log, it's 100% there. This is more of a DX improvement, so the user won't have to change to the specific mode in order to subscribe to invalid and it certainly works for some user who doesn't want to set it to onChange and still wants to subscribe to invalid. So this won't get revert back.

**Comment by 0x2509** [comment]
> Please refer to the release log, it's 100% there. This is more of a DX improvement, so the user won't have to change to the specific mode in order to subscribe to invalid and it certainly works for some user who doesn't want to set it to onChange and still wants to subscribe to invalid. So this won't get revert back.

Yeah i see it in the release logs. I mean the behaviour change should be documented in the docs (https://react-hook-form.com/) with 'isValid'.  

Thanks for your answer. We will try to live with it and find a solution for our problem :)

**Comment by bluebill1049** [comment]
> Yeah i see it in the release logs. I mean the behaviour change should be documented in the docs (https://react-hook-form.com/) with 'isValid'.

Sure, I can include a note in the doc after work today.
